index.js

```diff
@@ -30,7 +30,7 @@ function throwOnRelease () {
 
 function release (client, err) {
   client.release = throwOnRelease
-  if (err || this.ending) {
+  if (err || this.ending || !client._queryable || client._ending) {
```

Comment:
Alternatively re-adding a not queryable client could throw instead of silently re-adding.